xdsl/dialects/omp.py

```diff
@@ -593,6 +595,55 @@ def verify_(self) -> None:
         return super().verify_()
 
 
+@irdl_op_definition
+class SimdOp(IRDLOperation):
+    """
+    Implementation of upstream omp.simd
+    See external [documentation](https://mlir.llvm.org/docs/Dialects/OpenMPDialect/ODS/#ompsimd-ompsimdop).
+    """
+
+    name = "omp.simd"
+
+    ALIGN_COUNT: ClassVar = IntVarConstraint("ALIGN_COUNT", AnyInt())
+    LINEAR_COUNT: ClassVar = IntVarConstraint("LINEAR_COUNT", AnyInt())
+
+    aligned_vars = var_operand_def(
+        RangeOf(
+            AnyAttr(),  # TODO: OpenMP_PointerLikeTypeInterface
+            length=ALIGN_COUNT,
+        )
+    )
+    if_expr = opt_operand_def(i1)
+    linear_vars = var_operand_def(RangeOf(AnyAttr(), length=LINEAR_COUNT))
+    linear_step_vars = var_operand_def(RangeOf(eq(i32), length=LINEAR_COUNT))
+    nontemporal_vars = opt_operand_def()  # TODO: OpenMP_PointerLikeTypeInterface
+    private_vars = opt_operand_def()
+    reduction_vars = opt_operand_def()  # TODO: OpenMP_PointerLikeTypeInterface
+
+    alignments = opt_prop_def(
+        ArrayAttr.constr(RangeOf(base(IntegerAttr[i64]), length=ALIGN_COUNT))
+    )
+    order = opt_prop_def(OrderKindAttr)
+    private_syms = opt_prop_def(ArrayAttr[SymbolRefAttr])
+    reduction_syms = opt_prop_def(ArrayAttr[SymbolRefAttr])
+    simdlen = opt_prop_def(IntegerAttr[i64])
```

Comment:
```suggestion
    simdlen = opt_prop_def(IntegerAttr.constr(type=i64))
```
I don't think what you have written actually enforces that the type is i64. Should be the same with the other `IntegerAttr`s

Reply:
It seems to, if I try to do `simdlen = 64 : ui64` in a test I get `Operation does not verify: Expected attribute i64 but got ui64`

Reply:
Fair enough